pygsti/algorithms/germselection.py

```diff
@@ -2219,3 +2472,1515 @@ def rcl_fn(x): return _scoring.filter_composite_rcl(x, alpha)
     bestSoln = localSolns[_np.argmin(finalScores)]
 
     return (bestSoln, initialSolns, localSolns) if return_all else bestSoln
+
+
+def clean_germ_list(model, circuit_cache, eq_thresh= 1e-6):
+    #initialize an identity matrix of the appropriate dimension
+    
+    cleaned_circuit_cache= circuit_cache.copy()
+                   
+    
+    #remove circuits with duplicate PTMs
+    #The list of available fidcuials is typically
+    #generated in such a way to be listed in increasing order
+    #of depth, so if we search for dups in that order this should
+    #generally favor the shorted of a pair of duplicate PTMs.
+    #cleaned_cache_keys= list(cleaned_circuit_cache.keys())
+    #cleaned_cache_PTMs= list(cleaned_circuit_cache.values())
+    #len_cache= len(cleaned_cache_keys)
+    
+    #reverse the list so that the longer circuits are at the start and shorter
+    #at the end for better pop behavior.
+    
+    #TODO: add an option to partition the list into smaller chunks to dedupe
+    #separately before regrouping and deduping as a whole. Should be a good deal faster. 
+    
+    unseen_circs  = list(cleaned_circuit_cache.keys())
+    unseen_circs.reverse()
+    unique_circs  = []
+    
+    #While unseen_circs is not empty
+    while unseen_circs:
+        current_ckt = unseen_circs.pop()
+        current_ckt_PTM = cleaned_circuit_cache[current_ckt]
+        unique_circs.append(current_ckt)            
+        #now iterate through the remaining elements of the set of unseen circuits and remove any duplicates.
+        is_not_duplicate=[True]*len(unseen_circs)
+        for i, ckt in enumerate(unseen_circs):
+            #the default tolerance for allclose is probably fine.
+            if _np.linalg.norm(cleaned_circuit_cache[ckt]-current_ckt_PTM)<eq_thresh: #use same threshold as defined in the base find_fiducials function
+                is_not_duplicate[i]=False
+        #reset the set of unseen circuits.
+        unseen_circs=list(itertools.compress(unseen_circs, is_not_duplicate))
+    
+    #rebuild the circuit cache now that it has been de-duped:
+    cleaned_circuit_cache_1= {ckt_key: cleaned_circuit_cache[ckt_key] for ckt_key in unique_circs}
+        
+    #now that we've de-duped the circuit_cache, we can pull out the keys of cleaned_circuit_cache_1 to get the
+    #new list of available fiducials.
+    
+    cleaned_availableGermList= unique_circs
+    
+        
+    return cleaned_availableGermList, cleaned_circuit_cache_1
+    
+
+#new function for taking a list of available fiducials and generating a cache of the PTMs
+#this will also be useful trimming the list of effective identities and fiducials with
+#duplicated effects.
+
+def create_circuit_cache(model, circuit_list):
```

Comment:
I think is function is a duplicate of one in fiducialselection.py -- if so we should just use that one.

Reply:
Ah, I see.  You could just leave this until later, but it'd probably be useful to at least add a comment now mentioning that this is essentially a duplicate of what's in `fiducialselection.py`.